### PR TITLE
[prometheus-snmp-exporter] Fixes #398 and adds some features

### LINKED ISSUE #398: [prometheus-snmp-exporter] Multiple issues with the chart
**Describe the bug**
* The ServiceMonitor that is deployed by this chart uses wrong label to identify the right pods
* The ServiceMonitor would not work properly when multiple instances of the snmp_exporter are deployed

**Version of Helm and Kubernetes**:

Helm Version:

```console
$ helm version
version.BuildInfo{Version:"v3.4.0", GitCommit:"7090a89efc8a18f3d8178bf47d2462450349a004", GitTreeState:"dirty", GoVersion:"go1.15.3"}
```

Kubernetes Version:

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T18:49:28Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.7", GitCommit:"b4455102ef392bf7d594ef96b97a4caa79d729d9", GitTreeState:"clean", BuildDate:"2020-06-17T11:32:20Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**: prometheus-snmp-exporter

**Which version of the chart**: 
0.0.6

**What happened**: 
* When ServiceMonitor is deployed, no pods are matched to be collected
* When multiple instances of the exporter are deployed, none are collected

**What you expected to happen**:
* Metrics from the snmp_exporter are properly collected
* When multiple exporters are deployed, metrics are collected from all instances correctly

**How to reproduce it** (as minimally and precisely as possible):
**Changed values of values.yaml** (only put values which differ from the defaults):

values.yaml

```console
serviceMonitor:
  enabled: true
  namespace: mon
  params:
    enabled: true
    conf:
      module: 
        - fortigate_snmp
      target:
        - x.x.x.x
```

**The helm command that you execute and failing/misfunctioning**:

For example:

```console
helm install my-release prometheus-community/name-of-chart --version version --values values.yaml
```

**Helm values set after installation/upgrade:**

```console
helm get values my-release
```

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by ni-ajardan**
#### What this PR does / why we need it:
- Fixes ServiceMonitor to crawl the correct pods
- Fixes ServiceMonitor to properly work when multiple instances
  of the chart are installed
- Allows specifying relabelings and metricRelabelings configuration
  in ServiceMonitor

#### Which issue this PR fixes
  - fixes #398

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-snmp-exporter/README.md [hunk 1]
```diff
@@ -91,3 +91,19 @@ scrape_configs:
       - target_label: __address__
         replacement: my-service-name:9116  # The SNMP exporter's Service name and port.
 ```
+
+Eaxample configuration via a ServiceMonitor
```
Comment (monotek) [inline charts/prometheus-snmp-exporter/README.md:95]: imho the servicemonitor could be added as manifest and being enbaled / disabled by config value.
for example like here:
https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-rabbitmq-exporter/templates/servicemonitor.yaml

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-snmp-exporter/README.md (hunk 1 above)
Comment (ni-ajardan) [reply charts/prometheus-snmp-exporter/README.md:95]: Thank you for your commend @monotek. It is actually present here (I also made some fixes in that template down below), and can be enabled/disabled, even my example has the flag in it on line 98

File: charts/prometheus-snmp-exporter/Chart.yaml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 apiVersion: v1
 description: Prometheus SNMP Exporter
 name: prometheus-snmp-exporter
-version: 0.0.6
-appVersion: 0.14.0
+version: 0.0.7
```
Comment (monotek) [inline charts/prometheus-snmp-exporter/Chart.yaml:4]: feature update should raise minor version.
would be 0.1.0 in this case what should have been the first version anyway.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-snmp-exporter/Chart.yaml (hunk 2 above)
Comment (ni-ajardan) [reply charts/prometheus-snmp-exporter/Chart.yaml]: @monotek Thanks for the correction, should be fixed now :)

Comment (ni-ajardan) [reply charts/prometheus-snmp-exporter/Chart.yaml]: Lint already considers this 0.1.0, should I bump to 0.1.1 ?

Comment (monotek) [reply charts/prometheus-snmp-exporter/Chart.yaml]: Yes, please...

Comment (ni-ajardan) [reply charts/prometheus-snmp-exporter/Chart.yaml]: Done

**Review by monotek — CHANGES_REQUESTED** [review]
Please also sign your commits so the dco check passes.

**Review by monotek — APPROVED** [review]

**Comment by ni-ajardan** [comment]
Thank you !

**Pull request merged**